setup.py

```diff
@@ -32,18 +31,21 @@ def fetch_requirements():
     return reqs
 
 
-# https://packaging.python.org/guides/single-sourcing-package-version/
-def find_version(version_file_path):
-    with open(version_file_path) as version_file:
-        version_match = re.search(
-            r"^__version__ = ['\"]([^'\"]*)['\"]", version_file.read(), re.M
-        )
-        # The following is used to build release packages.
-        # Users should never use it.
-        suffix = os.getenv("XFORMERS_VERSION_SUFFIX", "")
-        if version_match:
-            return version_match.group(1) + suffix
-        raise RuntimeError("Unable to find version string.")
+if os.getenv("BUILD_VERSION"):
+    version = os.getenv("BUILD_VERSION")
+else:
+    version_txt = os.path.join(this_dir, "version.txt")
+    with open(version_txt) as f:
+        version = f.readline().strip()
```

Comment:
Would we want to also add the  git sha in here if we are building from source?
Something like what [torchvision does in here](https://github.com/pytorch/vision/blob/7d2de404372b0a77c5dec825c62f739e75a351ee/setup.py#L42-L43)
This way we are consistent between building with `BUILD_VERSION` (from CI?) and building locally.

Thoughts?